static/js/tools/map/chart_loader.tsx

```diff
@@ -291,102 +176,96 @@ export function ChartLoader(): JSX.Element {
       </div>
     );
   }
-  const facetInfo = getFacetInfo(
-    statVar.value,
-    rawData.allEnclosedPlaceStat,
-    rawData.metadataMap
-  );
 
-  const onPlay = async (callback: () => void) => {
-    setOnPlayCallback(() => callback);
-    const metaHash = statVar.value.metahash || BEST_AVAILABLE_METAHASH;
-    if (metaHash in sampleDatesChartData) {
-      callback();
-      return;
-    }
-    fetchData(
-      placeInfo.value,
-      statVar.value,
-      isLoading,
-      setRawData,
-      display.value.showTimeSlider,
-      allSampleDates,
-      allSampleDates[metaHash],
-      sampleDatesChartData,
-      setSampleDatesChartData
+  if (mapType === MAP_TYPE.D3 && chartStore.geoJson.error) {
+    return (
+      <div className="p-5">
+        {`Sorry, maps are not available for ${placeInfo.value.enclosedPlaceType} in ${placeInfo.value.selectedPlace.name}. Try picking another place or type of place.`}
+      </div>
     );
-  };
+  }
 
-  const updateDate = (date: string) => {
-    // Check if any data is fetched at all for the date
-    const metaHash = statVar.value.metahash || BEST_AVAILABLE_METAHASH;
-    let placeStatData = false;
-    if (
-      metaHash in sampleDatesChartData &&
-      date in sampleDatesChartData[metaHash]
-    ) {
-      for (const place in sampleDatesChartData[metaHash][date]
-        .enclosedPlaceStat) {
-        if (
-          sampleDatesChartData[metaHash][date].enclosedPlaceStat[place].value
-        ) {
-          placeStatData = true;
-          break;
-        }
-      }
-    }
-    statVar.setDate(date);
+  if (!_.isEqual(chartStore.mapValuesDates.context.statVar, statVar.value)) {
+    return null;
+  }
 
-    // Skip update if date has no data
-    if (
-      placeStatData ||
-      (metaHash !== BEST_AVAILABLE_METAHASH &&
-        sampleDatesChartData[metaHash][date].allEnclosedPlaceStat[metaHash])
-    ) {
-      loadChartData(
-        chartStore,
-        sampleDatesChartData[metaHash][date],
-        placeInfo.value,
-        allSampleDates,
-        statVar.value,
-        setChartData,
-        dispatchSources,
-        dispatchMetadata,
-        display
-      );
+  if (
+    !_.isEqual(chartStore.mapValuesDates.context.placeInfo, placeInfo.value)
+  ) {
+    console.log("map value placeInfo do not match");
+    return null;
+  }
+
+  const date = getDate(statVar.value.dcid, dateCtx.value);
+
+  let unit = "";
```

Comment:
maybe we can make a helper function for getUnit like we have for getDate and getRankingLink?